### PR TITLE
6672 arc_reclaim_thread() should use gethrtime() instead of ddi_get_lbolt()

### PR DESCRIPTION
**Pull request by grwilson**

### CODE REVIEW

File: usr/src/uts/common/sys/time.h [hunk 1]
```diff
@@ -243,6 +247,9 @@ struct itimerval32 {
 #define	MSEC2NSEC(m)	((hrtime_t)(m) * (NANOSEC / MILLISEC))
 #define	NSEC2MSEC(n)	((n) / (NANOSEC / MILLISEC))
 
+#define	NSEC2SEC(n)	((n) / (NANOSEC / SEC))
+#define	SEC2NSEC(m)	((hrtime_t)(m) * (NANOSEC / SEC))
```
Comment (jeffpc) [inline usr/src/uts/common/sys/time.h]: Please at least file a new bug (and include the bug number in the commit message).  This way it's easier to see that these showed up.  (Personally, I'd even like these to be a separate commit...I think it's easy enough to do, and it makes it easier for folks to backport what they want more easily.  It also allows the advocates to revert one piece without manually having to separate out the changes.)

File: usr/src/uts/common/fs/zfs/arc.c [hunk 2]
```diff
@@ -3424,8 +3424,8 @@ arc_reclaim_thread(void)
 			 * even if we aren't being signalled)
 			 */
 			CALLB_CPR_SAFE_BEGIN(&cpr);
-			(void) cv_timedwait(&arc_reclaim_thread_cv,
-			    &arc_reclaim_lock, ddi_get_lbolt() + hz);
+			(void) cv_timedwait_hires(&arc_reclaim_thread_cv,
+			    &arc_reclaim_lock, SEC2NSEC(1), MSEC2NSEC(1), 0);
```
Comment (jeffpc) [inline usr/src/uts/common/fs/zfs/arc.c]: Previously, you were waking up every 1s but with 10ms granularity (assuming `hires_tick` is not tweaked).  Now, you are using 1ms granularity.  While I don't think that should make a difference, it is different behavior none the less.  Have you noticed any actual difference?

Comment (rmustacc) [inline usr/src/uts/common/fs/zfs/arc.c]: For what it's worth, we've been running everything with hires ticks and haven't noticed any changes here or issues from more aggressive arc reclaim activity.

**Comment by rmustacc** [comment]
Changes look good to me. Thanks @grwilson!

File: usr/src/uts/common/sys/time.h (hunk 1 above)
Comment (grwilson) [inline usr/src/uts/common/sys/time.h]: I will file a new bug. This work was done by one of our interns so I'm trying not to split up his commit.

File: usr/src/uts/common/fs/zfs/arc.c (hunk 2 above)
Comment (grwilson) [inline usr/src/uts/common/fs/zfs/arc.c]: We have not noticed any difference and have been running this for the past 9 months. To minimize merge conflict with the compressed ARC changes that are coming next, I needed to integrate this change first.

Comment (jeffpc) [inline usr/src/uts/common/fs/zfs/arc.c]: That's fair. I was just curious if you noticed anything.

File: usr/src/uts/common/sys/time.h (hunk 1 above)
Comment (jeffpc) [inline usr/src/uts/common/sys/time.h]: Thanks. LGTM.

Comment (grwilson) [inline usr/src/uts/common/sys/time.h]: I filed https://www.illumos.org/issues/6673. Thanks for taking a look!

File: usr/src/uts/common/fs/zfs/arc.c (hunk 2 above)
Comment (richardelling) [inline usr/src/uts/common/fs/zfs/arc.c]: > On Feb 24, 2016, at 7:03 AM, Robert Mustacchi <notifications@github.com> wrote:
> 
> In usr/src/uts/common/fs/zfs/arc.c <https://github.com/openzfs/openzfs/pull/70#discussion_r53950489>:
> 
> > @@ -3424,8 +3424,8 @@ arc_reclaim_thread(void)
> >  			 * even if we aren't being signalled)
> >  			 */
> >  			CALLB_CPR_SAFE_BEGIN(&cpr);
> > -			(void) cv_timedwait(&arc_reclaim_thread_cv,
> > -			    &arc_reclaim_lock, ddi_get_lbolt() + hz);
> > +			(void) cv_timedwait_hires(&arc_reclaim_thread_cv,
> > +			    &arc_reclaim_lock, SEC2NSEC(1), MSEC2NSEC(1), 0);
> For what it's worth, we've been running everything with hires ticks and haven't noticed any changes here or issues from more aggressive arc reclaim activity.
> 
> 

ditto, I've been running hires for a few years now and do monitor arc reclaim activity continuously.
Over in the Solaris camp, they are tickless, but still suffer from reclaim issues through 11.1.
 -- richard

**Pull request closed without merging**